test/replication-luatest/gh_7592_local_write_with_syncro_test.lua

```diff
@@ -0,0 +1,93 @@
+local t = require('luatest')
+local server = require('test.luatest_helpers.server')
+local cluster = require('test.luatest_helpers.cluster')
+local fiber = require('fiber')
+
+local g = t.group('gh-7592')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new{}
+    local box_cfg = {
+        replication_synchro_timeout = 1000,
+        replication_synchro_quorum = 3,
+        replication_timeout = 0.1,
+        election_mode = 'off',
+        replication = {
+            server.build_instance_uri('master'),
+            server.build_instance_uri('replica'),
+        },
+    }
+    cg.master = cg.cluster:build_and_add_server{
+        alias = 'master',
+        box_cfg = box_cfg,
+    }
+    cg.replica = cg.cluster:build_and_add_server{
+        alias = 'replica',
+        box_cfg = box_cfg,
+    }
+    cg.cluster:start()
+    cg.master:exec(function()
+        box.ctl.promote()
+        box.schema.space.create('sync', {is_sync = true})
+        box.space.sync:create_index('pk')
+        box.schema.space.create('loc', {is_local = true})
+        box.space.loc:create_index('pk')
+    end)
+    cg.replica:wait_vclock_of(cg.master)
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+local function wait_synchro_queue_is_busy(server)
```

Comment:
> Besides, you don't really need this function at all. In its single usage case you could just do cg.replica:wait_vclock_ok(cg.master).

Not really. I'd also need to wait until lsn is bumped on master. So it would turn out uglier IMO.

Renamed the function though:
```diff
diff --git a/test/replication-luatest/gh_7592_local_write_with_syncro_test.lua b/test/replication-luatest/gh_7592_local_write_with_syncro_test.lua
index 8cddd4a3f..a2af41de9 100644
--- a/test/replication-luatest/gh_7592_local_write_with_syncro_test.lua
+++ b/test/replication-luatest/gh_7592_local_write_with_syncro_test.lua
@@ -40,7 +40,7 @@ g.after_each(function(cg)
     cg.cluster:drop()
 end)
 
-local function wait_synchro_queue_is_busy(server)
+local function wait_synchro_queue_has_entries(server)
     t.helpers.retrying({}, function()
         local len = server:exec(function()
             return box.info.synchro.queue.len
@@ -68,7 +68,7 @@ g.test_local_space_write_on_replica = function(cg)
         end)
     end)
     f:set_joinable(true)
-    wait_synchro_queue_is_busy(cg.replica)
+    wait_synchro_queue_has_entries(cg.replica)
     local fid = cg.replica:exec(function()
         local t = require('luatest')
         t.assert(box.info.synchro.queue.len == 1, 'Queue is filled')  
```